### PR TITLE
Move data dir inside the package

### PR DESCRIPTION
**Pull request by jamienoss**
Move ``flfm/data`` -> ``flfm/flfm/tests/data``

Note: The data isn't technically in the "package" as it will not get packaged up by ``setuptools``. We can get it to do this, I'm just not sure that we should, but am on the fence and could easily be swayed.

### CODE REVIEW

File: README.md
Comment (ryanhausen) [inline README.md:1]: Running the instructions to install from scratch in a new environment installs the CUDA version pytorch is looks like.

```conda create -n test python=3.11```
```python  -m pip install -e .```

```
Installing collected packages: sortedcontainers, nvidia-cusparselt-cu12, mpmath, zipp, zict, urllib3, typing-extensions, triton, tornado, toolz, termcolor, tblib, sympy, pyyaml, python-dotenv, psutil, pillow, packaging, opt_einsum, nvidia-nvtx-cu12, nvidia-nvjitlink-cu12, nvidia-nccl-cu12, nvidia-curand-cu12, nvidia-cufile-cu12, nvidia-cuda-runtime-cu12, nvidia-cuda-nvrtc-cu12, nvidia-cuda-cupti-cu12, nvidia-cublas-cu12, numpy, networkx, msgpack, MarkupSafe, locket, fsspec, filelock, cloudpickle, click, annotated-types, typing-inspection, scipy, pydantic-core, partd, nvidia-cusparse-cu12, nvidia-cufft-cu12, nvidia-cudnn-cu12, ml_dtypes, jinja2, importlib_metadata, fire, pydantic, nvidia-cusolver-cu12, jaxlib, dask, torch, pydantic-settings, jax, distributed, torchvision, torchaudio, flfm
Successfully installed MarkupSafe-3.0.2 annotated-types-0.7.0 click-8.1.8 cloudpickle-3.1.1 dask-2025.4.1 distributed-2025.4.1 filelock-3.18.0 fire-0.7.0 flfm-0.1.3.dev29+g27ee56d.d20250509 fsspec-2025.3.2 importlib_metadata-8.7.0 jax-0.6.0 jaxlib-0.6.0 jinja2-3.1.6 locket-1.0.0 ml_dtypes-0.5.1 mpmath-1.3.0 msgpack-1.1.0 networkx-3.4.2 numpy-2.1.3 nvidia-cublas-cu12-12.6.4.1 nvidia-cuda-cupti-cu12-12.6.80 nvidia-cuda-nvrtc-cu12-12.6.77 nvidia-cuda-runtime-cu12-12.6.77 nvidia-cudnn-cu12-9.5.1.17 nvidia-cufft-cu12-11.3.0.4 nvidia-cufile-cu12-1.11.1.6 nvidia-curand-cu12-10.3.7.77 nvidia-cusolver-cu12-11.7.1.2 nvidia-cusparse-cu12-12.5.4.2 nvidia-cusparselt-cu12-0.6.3 nvidia-nccl-cu12-2.26.2 nvidia-nvjitlink-cu12-12.6.85 nvidia-nvtx-cu12-12.6.77 opt_einsum-3.4.0 packaging-25.0 partd-1.4.2 pillow-11.2.1 psutil-7.0.0 pydantic-2.11.4 pydantic-core-2.33.2 pydantic-settings-2.9.1 python-dotenv-1.1.0 pyyaml-6.0.2 scipy-1.15.3 sortedcontainers-2.4.0 sympy-1.14.0 tblib-3.1.0 termcolor-3.1.0 toolz-1.0.0 torch-2.7.0 torchaudio-2.7.0 torchvision-0.22.0 tornado-6.4.2 triton-3.3.0 typing-extensions-4.13.2 typing-inspection-0.4.0 urllib3-2.4.0 zict-3.0.0 zipp-3.21.0
```

File: README.md [hunk 1]
```diff
@@ -74,9 +74,11 @@ Follow the above [Build with Python ecosystem instructions](#with-python-ecosyst
 
 Using the command line interface (i.e., from a terminal prompt):
 ```term
-python flfm/cli.py data/yale/light_field_image.tif data/yale/measured_psf.tif reconstructed_image.tiff --lens_radius=230 --lens_center="(1000,980)" --backend=torch
+python flfm/cli.py flfm/tests/data/yale/light_field_image.tif flfm/tests/data/yale/measured_psf.tif reconstructed_image.tiff --lens_radius=230 --lens_center="(1000,980)" --backend=torch
```
Comment (ryanhausen) [inline README.md:77]: You need to add the cli command here, e.g. `main` or `export`

File: docker/Dockerfile.java.pytorch [hunk 2]
```diff
@@ -38,7 +38,7 @@ FROM ubuntu/jre:21-24.04_stable
 WORKDIR /
 COPY --from=java_builder /flfm/target/flfm-0.1-jar-with-dependencies.jar /flfm/
 COPY --from=py_builder /data/model.pt /data/
-COPY ./data/yale/*.tif /data/
+COPY .flfm/tests/data/yale/*.tif /data/
```
Comment (ryanhausen) [inline docker/Dockerfile.java.pytorch:41]: Should this be `./flfm/tests/data/yale/*.tif /data/` (with the leading slash)?

File: notebooks/jax_flfm.ipynb [hunk 3]
```diff
@@ -38,8 +38,9 @@
     "    return jnp.stack(imgs, axis=0)\n",
     "\n",
     "\n",
-    "image = open_image(\"../data/yale/light_field_image.tif\")\n",
-    "measured_psf = open_image(\"../data/yale/measured_psf.tif\")\n",
+    "# NOTE: The below data files are only present when cloning the repo and not when pip installing the package.\n",
+    "image = open_image(\"tests/data/yale/light_field_image.tif\")\n",
```
Comment (ryanhausen) [inline notebooks/jax_flfm.ipynb:42]: I think these paths need call the util repo location or be slightly different, e.g.:

`../flfm/data/yale/light_field_image.tif`

File: notebooks/pytorch_flfm.ipynb [hunk 4]
```diff
@@ -63,8 +63,9 @@
     "    return torch.stack(imgs, dim=0)\n",
     "\n",
     "\n",
-    "image = open_image(\"../data/yale/light_field_image.tif\").to(device)\n",
-    "measured_psf = open_image(\"../data/yale/measured_psf.tif\").to(device)\n",
+    "# NOTE: The below data files are only present when cloning the repo and not when pip installing the package.\n",
+    "image = open_image(\"tests/data/yale/light_field_image.tif\").to(device)\n",
```
Comment (ryanhausen) [inline notebooks/pytorch_flfm.ipynb:67]: I think these paths need call the util repo location or be slightly different, e.g.:

`../flfm/data/yale/light_field_image.tif`

**Review by ryanhausen — CHANGES_REQUESTED** [review]
Thanks for putting this together!

A couple minor things and then I think it's ready!

File: README.md
Comment (jamienoss) [reply README.md:1]: This is unrelated to this PR and more related to #165. However, this behavior is to be expected as ``pip install`` uses the loose deps in ``pyproject.toml`` and not the requirements files that #165 addressed.

This should be handled by #171

**Review by ryanhausen — APPROVED** [review]
LGTM.

You committed the cell outputs in the notebook. I don't mind leaving them in, but I think you prefer to clear the notebook.

**Comment by jamienoss** [comment]
>You committed the cell outputs in the notebook. I don't mind leaving them in, but I think you prefer to clear the notebook.

No, you did, and I deleted them lol.

**Comment by ryanhausen** [comment]
oh man.... shame on me haha
